cmake/Platform/Core/BoardPropertiesReader.cmake

```diff
@@ -47,24 +47,28 @@ function(_recursively_replace_properties BOARD_ID PROPERTY_VALUE_VAR)
    # something like ${foo} is not matched as it could be a shell variable 
    # or a cmake variable or whatever, but not a Arduino property.
    #
-   while("${${PROPERTY_VALUE_VAR}}" MATCHES "[^\$]{([^}]*)}")
-      set(variable "${CMAKE_MATCH_1}")
+   while("${${PROPERTY_VALUE_VAR}}" MATCHES "(^|[^\$]){([^}]*)}")
+   
+      set(variable "${CMAKE_MATCH_2}")
       
       # The following regular expression checks if the property (variable) 
       # that was referenced is one of a board.
       #
-      if("${variable}" MATCHES "${BOARD_ID}\.([^}]*)")
-         _get_board_property(${BOARD_ID} ${CMAKE_MATCH_1} repl_string)
-      elseif(NOT "${${variable}}" STREQUAL "")
-         # If it's not a board property, we try to find the variable 
-         # at global scope.
-         #
-         set(repl_string "${${variable}}")
-      endif()
+      _try_get_board_property("${BOARD_ID}" "${variable}" repl_string)
       
-      if(NOT "${repl_string}" STREQUAL "")
-         string(REGEX REPLACE "{${variable}}" "${repl_string}" ${PROPERTY_VALUE_VAR} "${${PROPERTY_VALUE_VAR}}")
+      if("${repl_string}" STREQUAL "")
```

Comment:
Only if that fails, we check it the variable is defined at global scope, e.g. in platforms.txt.